File: src/services/BookingManagement/BookingManagementService.Domain/ShoppingCarts/Abstractions/PriceCalculationResult.cs
namespace CinemaTicketBooking.Domain.ShoppingCarts.Abstractions;

public record PriceCalculationResult (decimal TotalCartAmountBeforeDiscounts,
    decimal TotalCartAmountAfterDiscounts, 
    decimal TotalCartDiscounts, ICollection<AppliedPriceRule> AppliedPriceRules);
    
public record AppliedPriceRule( Guid Id ,decimal Discount);

File: src/clients/lib/src/shopping_carts/domain/entities/price_calculation_result.dart
import 'package:equatable/equatable.dart';

class PriceCalculationResult extends Equatable {
  final double? totalCartAmountBeforeDiscounts;
  final double? totalCartAmountAfterDiscounts;
  final double? totalCartDiscounts;

  const PriceCalculationResult({this.totalCartDiscounts, this.totalCartAmountBeforeDiscounts, this.totalCartAmountAfterDiscounts});

  @override
  // TODO: implement props
  List<Object?> get props => [totalCartDiscounts, totalCartAmountBeforeDiscounts, totalCartAmountAfterDiscounts];
}
